crates/oxc_semantic/src/scope.rs

```diff
@@ -173,6 +173,16 @@ impl ScopeTree {
         }
     }
 
+    pub fn change_parent_id(&mut self, scope_id: ScopeId, new_parent_id: Option<ScopeId>) {
+        let old_parent_id = self.parent_ids[scope_id];
+        if self.build_child_ids {
+            if let Some(old_parent_id) = old_parent_id {
+                self.child_ids[old_parent_id].retain(|&child_id| child_id != scope_id);
+            }
+        }
```

Comment:
I think need to record this scope as child of its new parent too.

```suggestion
        if self.build_child_ids {
            let old_parent_id = self.parent_ids[scope_id];
            if let Some(old_parent_id) = old_parent_id {
                self.child_ids[old_parent_id].retain(|&child_id| child_id != scope_id);
            }
            if let Some(new_parent_id) = new_parent_id {
                self.child_ids[new_parent_id].push(scope_id);
            }
        }
```

Reply:
Yes, it was handled in `self.set_parent_id`. But it causes two repeated branch checks for `self.build_child_ids`. I will refactor it so that it doesn't call `self.set_parent_id` anymore